frontend/src/features/admin/users/ManageUsersPage.tsx

```diff
@@ -98,20 +105,22 @@ export const ManageUsersPage = () => {
         value={filter}
         agencyLookups={agencies}
         rolesLookups={roles}
-        onChange={value => dispatch(getUsersFilterAction(value))}
+        onChange={value => {
```

Comment:
whenever the filter changes, update the filter in redux and reset the page index to 0.